alacritty_terminal/src/term/mod.rs

```diff
@@ -1707,18 +1717,21 @@ impl<T: EventListener> Handler for Term<T> {
                     *cell = bg.into();
                 }
 
+                self.grid[cursor.line][Column(0)].flags.insert(prompt_mark);
                 let range = Line(0)..=cursor.line;
                 self.selection = self.selection.take().filter(|s| !s.intersects_range(range));
             },
             ansi::ClearMode::Below => {
                 let cursor = self.grid.cursor.point;
+                let prompt_mark = self.grid[cursor.line][Column(0)].flags & Flags::PROMPT_MARK;
```

Comment:
Same here, question if we should keep the prompt mark.

Reply:
That's what zsh doing right after `precmd`. It calls `ClearBelow`. The `bash` and some other shells call `ClearLine::Right`. Though, we can remove the `above` one.